src/Native/Unix/System.Native/pal_termios.h

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+include "pal_types.h"
+
+/**
+ * Get states of serial line.
+ */
+extern "C" int32_t SystemNative_TermiosGetDcd(int fd);
```

Comment:
You can use `BEGIN_EXTERN_C` and `END_EXTERN_C` before and after the functions declarations, that will avoid you to repeat `extern "C"`.

Reply:
Or we can simply rename `pal_termios.cpp` to `pal_termios.c`, as there is no real dependency on libstdc++ in there, and we already have some C files in the codebase.

I know it was discussed previously, but I really see no harm either way. It's just a matter of picking dependencies sparingly for each build unit separately (just like I wouldn't add reference to a package in managed code unless I need it), otherwise one can always argue "we already depend on libstdc++ for 10 other reasons".

Reply:
You still need to be able to be included in C++ files, thus the need for `extern "C"`. My comment is only to help you reduce repetition, it's not a mandated or required change.